src/main/java/org/reaktivity/nukleus/tls/internal/stream/ClientStreamFactory.java

```diff
@@ -194,7 +194,7 @@ private MessageConsumer newAcceptStream(
 
             return applicationRef == route.sourceRef() &&
                     applicationName.equals(route.source().asString()) &&
-                    (hostname == null || Objects.equals(tlsHostname, hostname)) &&
+                    (tlsHostname == null || Objects.equals(tlsHostname, hostname)) &&
                     (applicationProtocol == null || Objects.equals(tlsApplicationProtocol, applicationProtocol));
```

Comment:
Why are the semantics for `tlsHostname` and `applicationProtocol` different ?
Perhaps the second one should be `tlsApplicationProtocol != null || ...` ?